client/v2/autocli/flag/map.go

```diff
@@ -202,11 +202,10 @@ func (m compositeMapType[T]) NewValue(ctx *context.Context, opts *Builder) Value
 func (m *compositeMapValue[T]) Set(s string) error {
```

Comment:
Could you add tests for this function to make sure we are not breaking functionality?